### PR TITLE
Add category exclusion filter

### PR DESCRIPTION
**Pull request by JoaRiski**
Add the option to exclude articles from selected categories from showing on Article List plugin

Refs [SER-3446](https://project.anders.fi/browse/SER-3446)

### CODE REVIEW

File: cmsplugin_articles_ai/cms_plugins.py [hunk 1]
```diff
@@ -13,12 +13,14 @@ class ArticleList(CMSPluginBase):
...
     render_template = "cmsplugin_articles_ai/article_lift_list.html"
-    fields = ["article_amount", "language_filter"]
+    fields = ["article_amount", "language_filter", "exclude_categories"]
 
     def get_articles(self, plugin_conf):
         return Article.objects.public(
             language=plugin_conf.language_filter,
-        ).filter(publisher_is_draft=get_draft_status())
+        ).filter(
+            publisher_is_draft=get_draft_status()
```
Comment (JsseL) [inline cmsplugin_articles_ai/cms_plugins.py:23]: Wondering why `public` doesn't already filter the drafts away. Should we fix that too?

**Review by JsseL — COMMENTED** [review]
Looks good to me. Though perhaps you could also add the tag excluding while at it?

File: cmsplugin_articles_ai/cms_plugins.py (hunk 1 above)
Comment (JoaRiski) [reply cmsplugin_articles_ai/cms_plugins.py:23]: I'd say out of scope here, though I was wondering the same. It's likely the editor view feature, as in, when you toggle CMS edit view it shows different (draft) articles in there than it does when viewing the live site (only published).

File: cmsplugin_articles_ai/models/plugin_models.py [hunk 2]
```diff
@@ -52,6 +52,17 @@ class ArticleListPlugin(CMSPlugin):
         verbose_name=_("tags"),
         related_name="article_list_plugins",
     )
+    exclude_categories = models.ManyToManyField(
+        Category,
+        verbose_name=("exclude categories"),
```
Comment (JsseL) [inline cmsplugin_articles_ai/models/plugin_models.py]: Nice parenthesis, but perhaps you meant `_(...)` and not just `(...)`

Comment (JoaRiski) [reply cmsplugin_articles_ai/models/plugin_models.py]: oh, heh, will fix

**Review by JsseL — APPROVED** [review]
Nice job @JoaRiski. Thanks for finetuning it!

**Comment by juhakujala** [comment]
I guess the tag excluding should not be a required field

screen shot 2017-09-28 at 9 46 43

**Comment by JoaRiski** [comment]
Good point.

**Pull request merged**